azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/domain/NetworkInterfaceCard.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecomputearm.domain;
+
+import com.google.auto.value.AutoValue;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+import java.util.List;
+import static com.google.common.collect.ImmutableList.copyOf;
+
+@AutoValue
+public abstract class NetworkInterfaceCard {
+
+
+    @AutoValue
+    public abstract static class NetworkInterfaceCardProperties{
+
+        public NetworkInterfaceCardProperties() {
+        }// For AutoValue only!
+
+        @Nullable
+        public abstract String provisioningState();
+
+        @Nullable
+        public abstract String resourceGuid();
+
```

Comment:
dnsSettings is needed only if we want to add support for custom DNS servers.
networkSecurityGroup should not be needed at this phase. We would need to add support for corresponding API as well: https://msdn.microsoft.com/en-us/library/azure/mt163656.aspx